### ISSUE TITLE
RFE: Allow overriding sssd.conf with secondary config file

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1268]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1264
- Created at 2012-03-19 12:55:20 by [sgallagh](https://pagure.io/user/sgallagh)
- Closed as Fixed
- Assigned to [dpal](https://pagure.io/user/dpal)
---

Quoted from https://fedorahosted.org/pipermail/sssd-devel/2012-March/009043.html

Wondering about whether this feature exists or is planned to exist? It is for use in a shared administration environment. We have central administration and local administrators. It would allow a central configuration for sssd with local changes. Puppet could manage the central main file and include a user managed portion. We handle sudoers in a similar way. I guess the main use of this might be to allow local admins to control who can login to a server. sssd.conf could include something like this:

    access_provider = simple
    simple_allow_groups = central_admins

and the included locally managed file could have

    simple_allow_users = user01, user03, user42

I know we can do this in other ways with puppet, but this would be simpler and it seems to me an include feature might have other uses.

### Comments

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-03-19 12:55:40](https://pagure.io/SSSD/sssd/issue/1264#232320)**

(In #1165) I think we need to divide this into two tickets. There really are two different requests going on here.

The original request for this ticket was "the possibility to split configuration based on a logic grouping of directives. A file per domain is a good example of what I was intending/asking."

However, the request made in the email thread was more like my second question. He wants the ability to set up the configuration centrally, but be able to merge "override" values into it from a secondary local file.

I think we probably need to account for both cases, but they deserve separate tickets. I've opened ticket #1264 to track this.

blockedby:  => 1165

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-03-22 14:16:41](https://pagure.io/SSSD/sssd/issue/1264#232321)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.11 beta
owner: somebody => dpal

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-03-22 14:18:39](https://pagure.io/SSSD/sssd/issue/1264#232322)**

Fields changed

rhbz:  => [https://bugzilla.redhat.com/show_bug.cgi?id=790113 790113]

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-08-16 13:28:19](https://pagure.io/SSSD/sssd/issue/1264#232323)**

Fields changed

proposed_priority:  => Nice to have

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-09-04 23:20:32](https://pagure.io/SSSD/sssd/issue/1264#232324)**

Moving all the features planned for 1.10 release into 1.10 beta.

milestone: SSSD 1.11 beta => SSSD 1.10 beta

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-09-04 23:50:13](https://pagure.io/SSSD/sssd/issue/1264#232325)**

Fields changed

priority: major => minor

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-12-20 23:33:42](https://pagure.io/SSSD/sssd/issue/1264#232326)**

Fields changed

selected:  => Not need

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-01-02 15:32:08](https://pagure.io/SSSD/sssd/issue/1264#232327)**

Moving tickets that are not a priority for SSSD 1.10 into the next release.

milestone: SSSD 1.10 beta => SSSD 1.11 beta

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-10-25 14:13:24](https://pagure.io/SSSD/sssd/issue/1264#232328)**

Fields changed

cc:  => Michael.Gliwinski@henderson-group.com
changelog:  => 
design:  => 
design_review:  => 0
fedora_test_page:  => 
review:  => 0

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-10-29 23:44:55](https://pagure.io/SSSD/sssd/issue/1264#232329)**

BTW ding-libs already support the required functionality to merge the configs so it is up to SSSD to define the merge rules and allow multiple configs.

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2014-02-14 02:35:56](https://pagure.io/SSSD/sssd/issue/1264#232330)**

This ticket is superseded by ticket #2247. Moving to Deferred. IMO #2247 when fully implemented would cover this use case too.

milestone: SSSD 1.13 beta => SSSD Deferred

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2016-11-17 12:00:20](https://pagure.io/SSSD/sssd/issue/1264#232331)**

This was implemented by config snippets.

mark:  => 0
resolution:  => fixed
sensitive:  => 0
status: new => closed

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2017-02-24 14:45:31](https://pagure.io/SSSD/sssd/issue/1264#232332)**

**Metadata Update from @sgallagh**:
- Issue assigned to dpal
- Issue marked as depending on: #1165
- Issue set to the milestone: SSSD Patches welcome
